eclipse-collections-api/src/main/java/org/eclipse/collections/api/list/MutableList.java

```diff
@@ -61,24 +63,43 @@
     MutableList<T> withoutAll(Iterable<? extends T> elements);
 
     @Override
-    MutableList<T> newEmpty();
+    default MutableList<T> newEmpty()
+    {
+        return Lists.mutable.empty();
+    }
 
     MutableList<T> clone();
 
     @Override
-    MutableList<T> tap(Procedure<? super T> procedure);
+    default MutableList<T> tap(Procedure<? super T> procedure)
+    {
+        this.forEach(procedure);
+        return this;
```

Comment:
While looking for methods that could be moved up, I found plenty like this one where it could already be moved up and doesn't need the new Lists factory at all.

Reply:
If you wouldn't mind, I can selectively pick these and create a different PR. Like @mohrezaei mentioned there are few unrelated fixes and we might lose them during rebasing, editing. If you have it cleanly segregated in your working tree then let us get these unrelated one merged.